src/framework/components/collision/component.js

```diff
@@ -46,8 +46,7 @@ import { Component } from '../component.js';
  * collision volume's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
  * @property {number} height The total height of the capsule, cylinder or cone-shaped collision
  * volume from tip to tip. Defaults to 2.
- * @property {Asset} asset The asset for the model of the mesh collision volume - can also be
- * an asset id. Defaults to null.
+ * @property {Asset} number The asset id for the model of the mesh collision volume. Defaults to null.
```

Comment:
Well, first thing is that I think you meant:
```javascript
 * @property {number} asset The asset id for the model of the mesh collision volume. Defaults to null.
```
But as it happens, I think the code should be changed to take an asset. I believe that's more consistent with the model component:

https://github.com/playcanvas/engine/blob/master/src/framework/components/model/component.js#L549-L551

Reply:
Usually components check if the value is a pc.Asset or an id and accept both.